Do not allow in-source builds

**Is your feature request related to a problem? Please describe.**
https://hsf-training.github.io/hsf-training-cmake-webpage/02-building/index.html

> Never do an “in-source” build - that is, run cmake . from the source directory. It will pollute your source directory with build outputs, CMake configuration files, and will disable out-of-source builds. A few packages do not allow the source directory to even sit inside the build directory; if that is the case, you need to change the relative path .. accordingly.

Just to clarify, you can point CMake at either the source directory from the build directory, or at an existing build directory from anywhere.

Currently it allows in source builds:
```
cmake .
-- The C compiler identification is GNU 8.4.0
-- The CXX compiler identification is GNU 8.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
...
```
the source directory gets polluted
```
git status
...
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        CMakeCache.txt
        CMakeFiles/
        CTestTestfile.cmake
        DartConfiguration.tcl
        Makefile
        cmake_install.cmake
```


**Describe the solution you'd like**
```
cmake .
CMake Error at CMakeLists.txt:4 (message):
  In-source builds not allowed.

          Please make a new directory (called a build directory) and run CMake from there.
          You may need to remove 'CMakeCache.txt' and 'CMakeFiles/'.


-- Configuring incomplete, errors occurred!
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,11 @@
 cmake_minimum_required(VERSION 3.1) 
+
+if ( ${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR} )
+        message(FATAL_ERROR "In-source builds not allowed.
+        Please make a new directory (called a build directory) and run CMake from there.
+        You may need to remove 'CMakeCache.txt' and 'CMakeFiles/'.")
+endif()
+
 project(MobilityDB VERSION 1.0)
 set(MOBILITYDB_VERSION 1.0)
 include(CheckSymbolExists)

